assets/js/frontend/course-video/youtube-extension.js

```diff
@@ -0,0 +1,79 @@
+( () => {
+	function handleVisibilityChange( player ) {
+		return function () {
+			if ( document.hidden ) {
+				player.pauseVideo();
+			}
+		};
+	}
+
+	function preventClick( event ) {
+		event.preventDefault();
+		return false;
+	}
+
+	function onYouTubePlayerStateChange( event ) {
+		const playerStatus = event.data;
+
+		if (
+			window.sensei.courseVideoSettings.courseVideoRequired &&
+			playerStatus === YT.PlayerState.ENDED
+		) {
+			document
+				.querySelectorAll( '[data-id="complete-lesson-button"]' )
+				.forEach( ( button ) => {
+					button.removeEventListener( 'click', preventClick );
+					button.disabled = false;
+				} );
+		}
+
+		if (
+			window.sensei.courseVideoSettings.courseVideoAutoComplete &&
+			playerStatus === YT.PlayerState.ENDED
+		) {
+			// submit complete lesson form
+			document
+				.querySelectorAll( '[data-id="complete-lesson-form"]' )
+				.forEach( ( form ) => {
+					form.submit();
```

Comment:
In this case, I think we could get only the first one with `querySelector`, right? Otherwise, we'll try to submit multiple forms at the same time.

Reply:
Fixed: https://github.com/Automattic/sensei/pull/4546/commits/cefe7de7ce916ac7c0b606613f5a417ae953f157